tests/managed/types.cs

```diff
@@ -80,3 +80,40 @@ public static class Type_String
 	public static string NonEmptyString { get { return "Hello World"; } }
 }
 
+#if NON_OBJC_SUPPORTED_TESTS
+
+public class BuiltinTypes
+{
+	public void ReturnsVoid () { }
+	public bool ReturnsBool () { return true; }
+	public sbyte ReturnsSByte () { return -5; }
+	public byte ReturnsByte () { return 5; }
+	public short ReturnsShort () { return -5; }
+	public ushort ReturnsUShort () { return 5; }
+	public int ReturnsInt () { return -5; }
+	public uint ReturnsUInt () { return 5; }
+	public long ReturnsLong () { return -5; }
+	public ulong ReturnsULong () { return 5; }
+	public char ReturnsChar () { return 'a'; }
+	public string ReturnsString () { return "Mono"; }
+
+	public bool PassAndReturnsBool (bool v) { return v; }
+	public sbyte PassAndReturnsSByte (sbyte v) { return v; }
+	public byte PassAndReturnsByte (byte v) { return v; }
+	public short PassAndReturnsShort (short v) { return v; }
+	public ushort PassAndReturnsUShort (ushort v) { return v; }
+	public int PassAndReturnsInt (int v) { return v; }
+	public uint PassAndReturnsUInt (uint v) { return v; }
+	public long PassAndReturnsLong (long v) { return v; }
+	public ulong PassAndReturnsULong (ulong v) { return v; }
+	public char PassAndReturnsChar (char v) { return v; }
+	public string PassAndReturnsString (string v) { return v; }
```

Comment:
all of the above should be working, they just need to be added in the Xcode unit tests (to confirm and avoid regressions)